generate_from_csv.py

```diff
@@ -359,71 +361,64 @@ def csv_to_dict(csv_path):
+    :type export: str or None
     :return: schedule of rotations
     :rtype: dict
     """
-    # add index
-    for i, row in enumerate(input):
-        row["index"] = i
-    # list of dicts to dict
-    input = {item['index']: item for item in input}
-
-    if not min_standing_time:
-        min_standing_time = 0
-
-    # filter for vehicle_type
-    vehicle_types = set(d['vehicle_type'] for d in input.values())
-    for vt in vehicle_types:
-        vt_line = {k: v for k, v in input.items() if v["vehicle_type"] == vt}
-        vehicle_number = 0
-        # sort list of vehicles by departure time
-        departures = {key: value for key, value in sorted(vt_line.items(),
-                      key=lambda x: x[1]['departure_time'])}
-        # sort list of vehicles by arrival time
-        arrivals = {key: value for key, value in sorted(vt_line.items(),
-                    key=lambda x: x[1]['arrival_time'])}
-        # get the first arrival
-        first_arrival_time = arrivals[list(arrivals.keys())[0]]["arrival_time"]
-        # parse through rotations in departure list and assign vehicle id
-        if min_standing_time == "None":
-            min_standing_time = 0
-        for rotation in departures.keys():
-            # as long as the first vehicle has not arrived yet: add vehicle_id number
-            if datetime.datetime.strptime(first_arrival_time, '%Y-%m-%d %H:%M:%S') + \
-                    datetime.timedelta(hours=min_standing_time) >= \
-                    datetime.datetime.strptime(vt_line[rotation]["departure_time"],
-                                               '%Y-%m-%d %H:%M:%S'):
-                vehicle_number += 1
-                input[rotation]["vehicle_id"] = vt + "_" + str(vehicle_number)
-            # as soon as departure is after the first arrival take first vehicle_number in line
-            elif datetime.datetime.strptime(arrivals[list(arrivals.keys())[0]]["arrival_time"],
-                                            '%Y-%m-%d %H:%M:%S') + \
-                    datetime.timedelta(hours=min_standing_time) < \
-                    datetime.datetime.strptime(vt_line[rotation]["departure_time"],
-                                               '%Y-%m-%d %H:%M:%S'):
-                arrival_rotation = list(arrivals.keys())[0]
-                a_bus_number = input[arrival_rotation]["vehicle_id"]
-                del arrivals[arrival_rotation]
-                input[rotation]["vehicle_id"] = a_bus_number
+    rotations_in_progress = []
+    idle_vehicles = []
+    vehicle_type_counts = {vehicle_type: 0 for vehicle_type in vehicle_types.keys()}
+
+    rotations = sorted(input, key=lambda d: d.get('departure_time'))
+
+    for rot in rotations:
+        # find vehicles that have completed rotation and stood for a minimum staning time
+        # mark those vehicle as idle
+        for r in rotations_in_progress:
+            # calculate min_standing_time deps
```

Comment:
what is "deps"? This should work in general, not only for busses.
```suggestion
            # calculate min_standing_time at a charging station for each vehicle
```